Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALChannelCalibrator.h

```diff
@@ -0,0 +1,128 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \class EMCALChannelCalibrator
+/// \brief  Perform the EMCAL bad channel calibration
+/// \author Hannah Bossi, Yale University
+/// \ingroup EMCALCalib
+/// \since Feb 11, 2021
+
+#ifndef EMCAL_CHANNEL_CALIBRATOR_H_
+#define EMCAL_CHANNEL_CALIBRATOR_H_
+
+#include "DetectorsCalibration/TimeSlotCalibration.h"
+#include "DetectorsCalibration/TimeSlot.h"
+#include "DataFormatsEMCAL/Cell.h"
+#include "EMCALBase/Geometry.h"
+#include "CCDB/CcdbObjectInfo.h"
+
+#include <array>
+#include <boost/histogram.hpp>
+
+namespace o2
+{
+namespace emcal
+{
+class EMCALChannelData
+{
+  //using Slot = o2::calibration::TimeSlot<o2::emcal::EMCALChannelData>;
+  using Cells = o2::emcal::Cell;
+  using boostHisto = boost::histogram::histogram<std::tuple<boost::histogram::axis::regular<double, boost::use_default, boost::use_default, boost::use_default>, boost::histogram::axis::integer<>>, boost::histogram::unlimited_storage<std::allocator<char>>>;
+
+ public:
+  // NCELLS includes DCal, treat as one calibration
+  o2::emcal::Geometry* mGeometry = o2::emcal::Geometry::GetInstanceFromRunNumber(300000);
+  int NCELLS = mGeometry->GetNCells();
```

Comment:
Should not be exposed as public data members. n general I doubt NCELLS is needed in the header file. The constructor below should maybe better go to the source file.